### PR TITLE
Upgrade Registration Service to the latest version

### LINKED ISSUE #266: Upgrade RegistrationService version in MVD
## Description
MVD is not compatible with the latest Registration Service changes.
 
## Acceptance Criteria

- [x] MVD is using the latest RS version
 
## Tasks
- [x] Update MVD with the latest Registration Service version
- [x] Fix issues in MVD

### PR DESCRIPTION
**Pull request by Izzzu**
## What this PR changes/adds

This PR upgrades Registration Service version to the latest upstream revision and fixes all the incompatibility issues:
- Adding vaults to local ald cloud set ups of Registration Service
- Using ParticipantDto instead of Participant
- Fix in the test

## Why it does that

MVD is not compatible with the latest Registration Service changes.

## Linked Issue(s)

Closes #266 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly?

### CODE REVIEW

File: .github/actions/gradle-setup/action.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ runs:
       with:
         repository: eclipse-dataspaceconnector/RegistrationService
         path: RegistrationService
-        ref: 0b39c0b0cb1884af58d829fa8396cfa3cf609117
+        ref: bbbfcec3c77dc74873844b7460d3dfc7017c68b6
```
Comment (cpeeyush) [inline .github/actions/gradle-setup/action.yml]: nit: latest updated due to recent PR merge.
```suggestion
        ref: a3b322d367ff76fcbd97d1df68d88480a6237bc5
```

**Review by marcgs — APPROVED** [review]

File: .github/workflows/destroy.yaml [hunk 2]
```diff
@@ -133,6 +133,9 @@ jobs:
           ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
           ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
 
+          # Passing dummy variables to terraform destroy, because destroy needs input variables to be defined, but uses the state.
+          TF_VAR_application_sp_client_secret: dummy
```
Comment (cpeeyush) [inline .github/workflows/destroy.yaml:137]: do we need this as this is already in .tfvars ?Asking as we do not need such dummy variable for anything else.

File: system-tests/docker-compose.yml [hunk 3]
```diff
@@ -152,9 +152,16 @@ services:
       WEB_HTTP_AUTHORITY_PORT: 8184
       WEB_HTTP_AUTHORITY_PATH: /authority
       EDC_IAM_DID_WEB_USE_HTTPS: "false"
+      EDC_CONNECTOR_NAME: registration-service
+      EDC_IDENTITY_DID_URL: did:web:did-server:registration-service
+      EDC_VAULT: /resources/vault/registration-service/registration-service-vault.properties
+      EDC_KEYSTORE: /resources/vault/registration-service/registration-service-keystore.jks
+      EDC_KEYSTORE_PASSWORD: test123
```
Comment (cpeeyush) [inline system-tests/docker-compose.yml:159]: nit: error verbose may also be added
```suggestion
      EDC_KEYSTORE_PASSWORD: test123
      EDC_ERROR_RESPONSE_VERBOSE: "true"
```

File: .github/workflows/destroy.yaml (hunk 2 above)
Comment (marcgs) [reply .github/workflows/destroy.yaml:137]: It is not in the .tfvars, as it contains sensitive data I think. Same pattern is applied for participants in line 85.

Comment (cpeeyush) [reply .github/workflows/destroy.yaml:137]: Right, oh I see it is secret. Resolving now...thanks

**Review by cpeeyush — APPROVED** [review]
Approved with suggestion

**Pull request merged**